### PR TITLE
SLSKPROTOCOL.md: Transfer Status Strings table and NS >2GB offset

### LINKED ISSUE #1985: Local file error (I/O error: cannot fit 'int' into an offset-sized integer)
Nicotine+ version: 3.2.2 (GTK 3.24.5).
Operating System/Distribution: Raspbian GNU/Linux 10 (buster) (armv7l).

### Describe the bug

I've started to get strange error messages some time ago, maybe after the latest Nicotine+ upgrade to the aforementioned version.

Some files can not be uploaded with the "Local file error" message in "Status" column of "Uploads" tab. At the same time a couple of error items appear in the log history: "I/O error: cannot fit 'int' into an offset-sized integer". I guess those error messages seem to be related one with another.

As I click the "Retry" option from the context menu for the specific uploading file, it disappears from the uploading queue.

The only feature to narrow the scope of the issue that I may provide is the following. Those files that I've seen this issue about have a pretty large size (the latest file with this error has more than 3 Gb of file size).

### Expected behavior

I guess there should be no such errors, no matter the size of any certain file being uploaded.

### Steps to reproduce the bug

I'm not sure about the exact steps to reproduce such an issue, but it occurres from time to time with some particular files.

### Additional context

I'm not sure how to provide some additional technical data for further investigation, so I'd be glad to add some additional feedback, in case it's possible.

### PR DESCRIPTION
**Pull request by slook**
Closes #1985

> It would be a good idea to add common transfer status messages to the Soulseek protocol documentation.

Please feel free to suggest what headers and content is required to populate the new table, or take it over as you deem appropriate.

### CODE REVIEW

File: doc/SLSKPROTOCOL.md [hunk 1]
```diff
@@ -73,6 +73,31 @@ If you find any inconsistencies, errors or omissions in the documentation, pleas
 | 0    | Download from Peer |
 | 1    | Upload to Peer     |
 
+### Transfer Status Strings
+
+| String                | Comments                                    |
+| --------------------- | ------------------------------------------- |
+| Queued                |                                             |
+| Transferring          |                                             |
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:81]: Not a protocol message.

File: doc/SLSKPROTOCOL.md [hunk 2]
```diff
@@ -73,6 +73,31 @@ If you find any inconsistencies, errors or omissions in the documentation, pleas
...
+| Connection timeout    |                                             |
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:82]: Not a protocol message.

File: doc/SLSKPROTOCOL.md [hunk 3]
```diff
@@ -73,6 +73,31 @@ If you find any inconsistencies, errors or omissions in the documentation, pleas
...
+| Pending shutdown.     |                                             |
+| User logged off       |                                             |
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:84]: Not a protocol message.

File: doc/SLSKPROTOCOL.md [hunk 4]
```diff
@@ -73,6 +73,31 @@ If you find any inconsistencies, errors or omissions in the documentation, pleas
...
+| Disallowed extension  | Sent by Soulseek NS for filtered extensions |
+| Aborted               |                                             |
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:86]: Not a protocol message.

File: doc/SLSKPROTOCOL.md [hunk 5]
```diff
@@ -73,6 +73,31 @@ If you find any inconsistencies, errors or omissions in the documentation, pleas
...
+| Cancelled             |                                             |
+| Paused                |                                             |
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:88]: Not a protocol message.

File: doc/SLSKPROTOCOL.md [hunk 6]
```diff
@@ -73,6 +73,31 @@ If you find any inconsistencies, errors or omissions in the documentation, pleas
...
+| Finished              |                                             |
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:89]: Protocol message is "Complete".

File: doc/SLSKPROTOCOL.md [hunk 7]
```diff
@@ -73,6 +73,31 @@ If you find any inconsistencies, errors or omissions in the documentation, pleas
...
+| Filtered              |                                             |
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:90]: Not a protocol message.

File: doc/SLSKPROTOCOL.md [hunk 8]
```diff
@@ -73,6 +73,31 @@ If you find any inconsistencies, errors or omissions in the documentation, pleas
...
+| Banned                |                                             |
+| Blocked country       |                                             |
+| Too many files        |                                             |
+| Too many megabytes    |                                             |
+| File not shared       |                                             |
+| File not shared.      | Newer variant containing a dot              |
+| Download folder error |                                             |
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:97]: Not a protocol message.

File: doc/SLSKPROTOCOL.md [hunk 9]
```diff
@@ -73,6 +73,31 @@ If you find any inconsistencies, errors or omissions in the documentation, pleas
...
+| Local file error      | Usually disk I/O, or Soulseek NS files >2GB |
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:98]: Not a protocol message.

**Comment by slook** [comment]
I suppose we could have an extra column(s) for "Usage" and "Direction" or something, since it is not very obvious in every case which messages are supposed to be sent by which peer.

**Pull request merged**